gapic-common/lib/gapic/grpc/service_stub.rb

```diff
@@ -49,22 +54,43 @@ class ServiceStub
     #     provided as a `::GRPC::Core::Channel`.)
     # @param interceptors [Array<::GRPC::ClientInterceptor>] An array of {::GRPC::ClientInterceptor} objects that will
     #   be used for intercepting calls before they are executed Interceptors are an EXPERIMENTAL API.
+    # @param channel_pool_config [::Gapic::ServiceStub:ChannelPool::Configuration] The configuration for channel
+    #     pool. (This argument is ignored when `credentials` is provided as a `::GRPC::Core::Channel`.)
```

Comment:
Consider instead raising ArgumentError if a channel is provided for `credentials` and `channel_pool_config` is set, instead of silently ignoring the latter. This helps users avoid bugs.

Reply:
Switched to raising `ArgumentError`.